util/fetch/src/client.rs

```diff
@@ -169,12 +170,42 @@ impl Drop for Client {
 }
 
 impl Client {
+	/// Create a new fetch client with a proxy.
+	pub fn new_with_proxy(https_proxy_url: Option<String>) -> Result<Self, Error> {
```

Comment:
A bit of code duplication. Is there a better way?

The problem I see here is the background_thread that is started during Client creation, and we can't add `set_proxy()` here, because `https_connector` with a [possible] proxy should be already created, during the thread start.

That's why I've added a separate factory method, to build a Client, but passing a Proxy URL into the Thread creation function.

Reply:
I decided to add new `new_with_proxy` because changing `new` definition will break current usages, e.g. in tests. But if you think it's ok, then I'll gladly change it.